servant-server/src/Servant/Server/Internal/ErrorFormatter.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE DataKinds #-}
+
+-- | TODO: more documentation for this module
+module Servant.Server.Internal.ErrorFormatter
+  where
+
+import           Data.String.Conversions
+                 (cs)
+import           Data.Typeable
+
+import           Servant.Server.Internal.Context
+import           Servant.Server.Internal.ServerError
+
+-- | 'Context' that contains default formatters for all error types.
+--
+-- Default formatters will just return HTTP 400 status code with error
+-- message as response body.
+type DefaultErrorFormatters = '[BodyParseErrorFormatter, URLParseErrorFormatter, HeaderParseErrorFormatter]
+
+defaultErrorFormatters :: Context DefaultErrorFormatters
+defaultErrorFormatters =
+  defaulyBodyParseErrorFormatter
+  :. defaultURLParseErrorFormatter
+  :. defaultHeaderParseErrorFormatter
+  :. EmptyContext
```

Comment:
What about `404 not found`?

Reply:
As far as I've seen, 404 is usually bodyless, so does not require customization.

But probably you and other maintainers know better. If you think 404 should be customizable as well, I'll add it.

Reply:
I think 404 might be (for people doing server-rendered HTML) one of the most important ones to customize! I think ideally every servant error should be customizable.

Reply:
Okay! I didn't think that usecase was still popular :)

Reply:
Done.